lib/api/fetchWithAuth.ts

```
@@ -0,0 +1,76 @@
+      } else {
+        resolve(false);
+      }
+    } catch (e) {
+      resolve(false);
+    } finally {
+      isRefreshing = false;
+      refreshPromise = null;
+    }
+  });
+
+  return refreshPromise;
```

Biome (2.1.2)

[error] Promise executor functions should not be `async`.



(lint/suspicious/noAsyncPromiseExecutor)